### PR TITLE
Flutter Response Model with Refactored SDK

### PR DESCRIPTION
**Pull request by lohanidamodar**

### CODE REVIEW

**Review by abnegate — COMMENTED** [review]
Where there's a list type, what do you think about (and would it be possible without coupling with Appwrite) removing the wrapper, e.g. returning `Future<List<Document>>` directly instead of `Future<DocumentList>`? It seems redundant since we can get `sum` from the languages `array` type (`.size()`, `.length`, `.Count` etc.)

File: src/Spec/Swagger2.php [hunk 1]
```diff
@@ -146,6 +146,17 @@ public function getMethods($service)
                         }
                     }
 
+                    $responses = $method['responses'];
+                    $responseModel = '';
+                    foreach($responses as $code => $desc) {
+                        if(isset($desc['schema']) && isset($desc['schema']['$ref'])) {
+                            $responseModel = $desc['schema']['$ref'];
+                            if(!empty($responseModel)) {
+                                $responseModel = substr($responseModel,14);
```
[the remark this answers is not in the document]
Comment (TorstenDittmann) [reply src/Spec/Swagger2.php:155]: Maybe we should do something like `str_replace('#/definitions/', '', $responseModel)` for better readability 👍🏻

File: templates/flutter/lib/src/models/model.dart.twig [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+{% if definition.additionalProperties %}
+            "data": data,
+{% endif %}
+        };
+    }
+
+{% if definition.additionalProperties %}
+    T convertTo<T>(T Function(Map) fromJson) => fromJson(data);
+{% endif %}
+{% if definition.name == 'documentList' %}
```
[the remark this answers is not in the document]
Comment (lohanidamodar) [reply templates/flutter/lib/src/models/model.dart.twig:46]: @TorstenDittmann @eldadfux any suggestions, 
here, if a schema is a List of sub schema and sub schema has additional properties then I want this to happen (right now only `documentList` meets this criteria.

[the remark this answers is not in the document]
Comment (TorstenDittmann) [reply templates/flutter/lib/src/models/model.dart.twig:46]: How does this criteria reflect on the specs?

[the remark this answers is not in the document]
Comment (lohanidamodar) [reply templates/flutter/lib/src/models/model.dart.twig:46]: for `document` it has `additionalProperties` true, but for `documentList` we don't have anything related in swagger

**Pull request merged**